empire.go

```diff
@@ -141,6 +145,10 @@ func (opts CreateOpts) Event() CreateEvent {
 
 // Create creates a new app.
 func (e *Empire) Create(ctx context.Context, opts CreateOpts) (*App, error) {
+	if err := e.MaintenanceMode; err != nil {
```

Comment:
One advantage of the decorator approach is we could generalize it a bit as an ACLEmpire and re-use it for access control. Something along the lines of:

```go
type Permissions map[string]error

type ACLEmpire struct {
        perms Permissions
        empire *Empire
}

func NewMaintenanceModeEmpire(err *MaintenanceMode, e *Empire) *ACLEmpire {
        return &ACLEmpire{
                empire: e,
                perms: Permissions{
                        "Create": err,
                },
        }
}

func (e *ACLEmpire) Create(ctx context.Context, opts CreateOpts) (*App, error) {
        if err := e.perms["Create"]; err != nil {
                return nil, err
        }

        return e.empire.Create(ctx, opts)
}
```